msticpy/data/uploaders/splunk_uploader.py

```diff
@@ -0,0 +1,252 @@
+
+        """
+        host = kwargs.get("host", None)
+        create_idx = kwargs.get("create_index", "False")
+        if not isinstance(data, pd.DataFrame):
+            raise MsticpyUserError(
+                "Data must be in Pandas DataFrame format.",
+                title="incorrect data format",
+            )
+        self._post_data(
+            data=data,
+            table_name=table_name,
+            index_name=index_name,
+            create_index=create_idx,
+            host=host,
+        )
+
+    def upload_file(  # type: ignore
+        self,
+        file_path: str,
+        index_name: str,
+        table_name: str = None,
+        delim: str = ",",
+        **kwargs,
+    ):
+        """
+        Upload a seperated value file to Splunk.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to the file to upload.
+        index_name : str
+            Name of the Splunk Index to add data to, if it doesn't exist it will be created.
+        table_name : str, optional
+            The souretype in Splunk data will be uploaded to, if not set the file name will be used.
+        delim : str, optional
+            Seperator value in file, by default ","
+        host : str, optional
+            Host name to upload data with, default will be 'Upload'
+
+        """
+        host = kwargs.get("host", None)
+        create_idx = kwargs.get("create_index", "False")
+        path = Path(file_path)
+        try:
+            data = pd.read_csv(path, delimiter=delim)
+        except (ParserError, UnicodeDecodeError):
+            raise MsticpyUserError(
+                "The file specified is not a seperated value file.",
+                "Incorrect file type.",
+            )
+
+        if not table_name:
+            table_name = str(path).split("\\")[-1].split(".")[0]
+        self._post_data(
+            data=data,
+            table_name=table_name,
+            index_name=index_name,
+            host=host,
+            create_index=create_idx,
+        )
+
+    def upload_folder(  # type: ignore
+        self,
+        folder_path: str,
+        index_name: str,
+        table_name: str = None,
+        delim: str = ",",
+        **kwargs,
+    ):
+        """
+        Upload all files in a folder to Splunk.
+
+        Parameters
+        ----------
+        folder_path : str
+            Path to folder to upload.
+        index_name : str
+            Name of the Splunk Index to add data to, if it doesn't exist it will be created.
+        table_name : str, optional
+            The souretype in Splunk data will be uploaded to, if not set the file name will be used.
+        delim : str, optional
+            Seperator value in files, by default ","
+        host : str, optional
+            Host name to upload data with, default will be 'Upload'
+
+        """
+        host = kwargs.get("host", None)
+        create_idx = kwargs.get("create_index", "False")
+        if delim != ",":
```

Comment:
Maybe it would be safer to have the user specify the glob pattern - you could just default to "*" if nothing is supplied. It would make it difficult to upload csvs that have a different extension.